src/mds/MDCache.cc

```diff
@@ -10076,11 +10076,6 @@ void MDCache::discover_path(CDir *base,
       onfinish = new C_MDC_RetryDiscoverPath2(this, base, snap, want_path);
     base->add_waiter(CDir::WAIT_SINGLEAUTH, onfinish);
     return;
-  } else if (from == mds->get_nodeid()) {
-    MDSContext::vec finished;
-    base->take_sub_waiting(finished);
-    mds->queue_waiters(finished);
```

Comment:
Is it appropriate to remove this though (queing of the required waiters)? I'm a bit unsure about this. Roping @batrick for his comments.